services/121-service/src/payments/fsp-integration/safaricom/services/safaricom.api.service.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { TokenSet } from 'openid-client';
+
+import { AuthResponseSafaricomApiDto } from '@121-service/src/payments/fsp-integration/safaricom/dtos/safaricom-api/auth-response-safaricom-api.dto';
+import { TransferResponseSafaricomApiDto } from '@121-service/src/payments/fsp-integration/safaricom/dtos/safaricom-api/transfer-response-safaricom-api.dto';
+import { SafaricomApiService } from '@121-service/src/payments/fsp-integration/safaricom/services/safaricom.api.service';
+import { CustomHttpService } from '@121-service/src/shared/services/custom-http.service';
+import { TokenValidationService } from '@121-service/src/utils/token/token-validation.service';
+
+const transferInput = {
+  transferAmount: 10,
+  phoneNumber: 'mocked-phone-number',
+  idNumber: 'mocked-id-number',
+  originatorConversationId: 'mocked-originator-conversation-id',
+};
+
+const mockPostTransferResponse: TransferResponseSafaricomApiDto = {
+  data: {
+    ConversationID: 'mocked-conversation-id',
+    OriginatorConversationID: 'mocked-originator-conversation-id',
+    ResponseCode: '0',
+    ResponseDescription: 'Success',
+  },
+};
+
+describe('SafaricomApiService', () => {
+  let safaricomApiService: SafaricomApiService;
+  let customHttpService: CustomHttpService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        SafaricomApiService,
+        TokenValidationService,
+        {
+          provide: CustomHttpService,
+          useValue: {
+            get: jest.fn(),
+            post: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    safaricomApiService = module.get<SafaricomApiService>(SafaricomApiService);
+    customHttpService = module.get<CustomHttpService>(CustomHttpService);
+
+    (customHttpService.post as jest.Mock).mockResolvedValue(
+      mockPostTransferResponse,
+    );
+  });
+
+  describe('transfer', () => {
```

Comment:
Is this "happy/typical path" that would be covered in Integration Test as well, and therefore can be left out of unit testing, depending on effort vs. added value?

Reply:
proposal is to remove these unit test and instead create extra integration test which calls payment endpoint twice, where checking that the 2nd time the outgoing GET authenticate call is not made. Such an assertion is new in our api-test portfolio, check with others if that is feasible first.

Reply:
talked about this with @RubenGeo and he agrees that api test does not serve this purpose well, as it's oriented at input and output of our api endpoints, and not on what happens inside. Instead look at options again to unit test this.

Reply:
I am missing some context to give a proper opinion on this, and am struggling to garner it just from reading the code. But perhaps it's ok if I don't share my opinion on this one and avoid "too many cooks spoiling the broth". If you really feel like my input could be valuable, feel free to put something in my calendar so maybe we can talk through it.